src/coredump/coredump.c

```diff
@@ -1016,45 +1016,41 @@ static int process_special_crash(const char *context[], int input_fd) {
         return 0;
 }
 
-static int process_kernel(int argc, char* argv[]) {
-
-        /* The small core field we allocate on the stack, to keep things simple */
-        char
-                *core_pid = NULL, *core_uid = NULL, *core_gid = NULL, *core_signal = NULL,
-                *core_session = NULL, *core_exe = NULL, *core_comm = NULL, *core_cmdline = NULL,
-                *core_cgroup = NULL, *core_cwd = NULL, *core_root = NULL, *core_unit = NULL,
-                *core_user_unit = NULL, *core_slice = NULL, *core_timestamp = NULL, *core_rlimit = NULL;
-
-        /* The larger ones we allocate on the heap */
-        _cleanup_free_ char
-                *core_owner_uid = NULL, *core_open_fds = NULL, *core_proc_status = NULL,
-                *core_proc_maps = NULL, *core_proc_limits = NULL, *core_proc_cgroup = NULL, *core_environ = NULL,
-                *core_proc_mountinfo = NULL, *core_container_cmdline = NULL;
+#define SET_IOVEC_FIELD(iovec, n_iovec, field, value)                   \
+        ({                                                              \
+                const char *_x = strappend((field), (value));           \
+                if (_x)                                                 \
+                        IOVEC_SET_STRING(iovec[(*(n_iovec))++], _x);    \
+                _x;                                                     \
+        })
```

Comment:
hmm, not fan of such macros i must say. Can't this be a regular function?

Also, this allocates memory and returns it as a "const char*"... That looks strange. In C free() expects a non-const ptr really..